apps/shared/avifjpeg.c

```diff
@@ -1062,8 +1073,13 @@ static avifBool avifJPEGReadInternal(FILE * f,
                 //   "offset of this portion as a 32-bit unsigned integer"
                 const uint32_t extendedXMPOffset = avifJPEGReadUint32BigEndian(
                     &marker->data[AVIF_JPEG_EXTENDED_XMP_TAG_LENGTH + AVIF_JPEG_EXTENDED_XMP_GUID_LENGTH + 4]);
-                if (((uint64_t)standardXMPSize + totalExtendedXMPSize) > SIZE_MAX) {
-                    fprintf(stderr, "XMP extraction failed: total XMP size is too large\n");
+                if (((uint64_t)standardXMPSize + totalExtendedXMPSize) > imageSizeLimit) {
```

Comment:
In the original code, this check compares the sum with `SIZE_MAX`. I believe it was intended to check if it would be safe to cast the sum to `size_t` in line 1101 below.

In the new code, this check is stricter, so the `size_t` cast is still safe.

Reply:
Optional: It may be good to add an assertion here:

```
    assert(imageSizeLimit <= SIZE_MAX);
```

Reply:
> Optional: It may be good to add an assertion here:
> 
> ```
>     assert(imageSizeLimit <= SIZE_MAX);
> ```

My IDE CLion shows the following warning:

```
Result of comparison of constant 18446744073709551615 with expression of type 'uint32_t' (aka 'unsigned int') is always true
```

Reply:
Kept the old condition in https://github.com/AOMediaCodec/libavif/pull/2047 instead